packages/tests/custom-elements/html/Element/toggleAttribute.test.html

```diff
@@ -0,0 +1,221 @@
+                this.attrCallbackArgs = [];
+              }
+
+              connectedCallback() {
+                this.connectedCallbackCount++;
+              }
+
+              disconnectedCallback() {
+                this.disconnectedCallbackCount++;
+              }
+
+              attributeChangedCallback(name, oldValue, newValue, namespace) {
+                this.attrCallbackArgs.push(
+                  Array.prototype.slice.apply(arguments)
+                );
+              }
+            }
+          );
+        }
+
+        suite('Toggling an unset attribute.', () => {
+          let localName;
+
+          setup(() => {
+            localName = generateLocalName();
+            defineWithLocalName(localName, ['attr']);
+          });
+
+          test('Toggling an attribute with no value (null) adds the attribute and triggers a callback.', () => {
+            const element = document.createElement(localName);
+
+            element.toggleAttribute('attr');
+
+            assert.equal(element.getAttribute('attr'), '');
+            assert.equal(element.attrCallbackArgs.length, 1);
+            assert.deepEqual(element.attrCallbackArgs[0], [
+              'attr',
+              null,
+              '',
+              null,
+            ]);
+          });
+
+          test('Toggling (force: true) an attribute with no value (null) adds the attribute and triggers a callback.', () => {
+            const element = document.createElement(localName);
+
+            element.toggleAttribute('attr', true);
+
+            assert.equal(element.getAttribute('attr'), '');
+            assert.equal(element.attrCallbackArgs.length, 1);
+            assert.deepEqual(element.attrCallbackArgs[0], [
+              'attr',
+              null,
+              '',
+              null,
+            ]);
+          });
+
+          test('Toggling (force: false) an attribute with no value (null) does not trigger a callback.', () => {
+            const element = document.createElement(localName);
+
+            element.toggleAttribute('attr', false);
+
+            assert.equal(element.getAttribute('attr'), null);
+            assert.equal(element.attrCallbackArgs.length, 0);
+          });
+        });
+
+        suite('Toggling a set attribute.', () => {
+          let localName1;
+          let localName2;
+
+          setup(() => {
+            localName1 = generateLocalName();
+            defineWithLocalName(localName1, []);
+            localName2 = generateLocalName();
+            defineWithLocalName(localName2, ['attr']);
+          });
+
+          test('Toggling an unobserved attribute removes the attribute but does not trigger a callback.', () => {
+            const element = document.createElement(localName1);
+
+            assert.equal(element.attrCallbackArgs.length, 0);
+
+            element.setAttribute('attr', 'abc');
+
+            assert.equal(element.attrCallbackArgs.length, 0);
```

Comment:
```suggestion
            assert.equal(element.getAttribute('attr'), 'abc');
            assert.equal(element.attrCallbackArgs.length, 0);
```
Maybe this is overkill but I kinda like confirming the state right before doing the thing in question when it relies on earlier stuff in the tests.